### PR TITLE
[Merged by Bors] - feat: miscellaneous lemmas about properties of `log` and `rpow`

### PR DESCRIPTION
**Pull request by dupuisf**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/Log/Basic.lean [hunk 1]
```diff
@@ -417,6 +417,12 @@ theorem isLittleO_log_id_atTop : log =o[atTop] id :=
   isLittleO_pow_log_id_atTop.congr_left fun _ => pow_one _
 #align real.is_o_log_id_at_top Real.isLittleO_log_id_atTop
 
+theorem isLittleO_const_log_atTop {c : ℝ} (hc : c ≠ 0) : (fun _ => c) =o[atTop] log := by
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/Log/Basic.lean:420]: You don't need `c ≠ 0` here, see `isLittleO_const_left`.

File: Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean [hunk 2]
```diff
@@ -601,6 +609,20 @@ theorem deriv_rpow_const (hf : DifferentiableAt ℝ f x) (hx : f x ≠ 0 ∨ 1 
   (hf.hasDerivAt.rpow_const hx).deriv
 #align deriv_rpow_const deriv_rpow_const
 
+lemma isBigO_deriv_rpow_const_atTop (p : ℝ) :
+    deriv (fun (x:ℝ) => x ^ p) =O[atTop] fun x => x ^ (p-1) := by
+  by_cases hp : p = 0
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean:614]: `rcases eq_or_ne p 0 with rfl | hp`? Also, for `p ≠ 0` you can have `IsTheta`.

File: Mathlib/Analysis/SpecialFunctions/Log/Basic.lean (hunk 1 above)
Comment (dupuisf) [reply Mathlib/Analysis/SpecialFunctions/Log/Basic.lean:420]: Indeed, not sure why I even had it there.

File: Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean (hunk 2 above)
Comment (dupuisf) [reply Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean]: I did this and extracted the `p ≠ 0` case as a separate lemma where we get `IsTheta`.

File: Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean [hunk 3]
```diff
@@ -601,6 +609,22 @@ theorem deriv_rpow_const (hf : DifferentiableAt ℝ f x) (hx : f x ≠ 0 ∨ 1 
   (hf.hasDerivAt.rpow_const hx).deriv
 #align deriv_rpow_const deriv_rpow_const
 
+lemma isTheta_deriv_rpow_const_atTop {p : ℝ} (hp : p ≠ 0) :
+    deriv (fun (x:ℝ) => x ^ p) =Θ[atTop] fun x => x ^ (p-1) := by
+  calc deriv (fun (x:ℝ) => x ^ p) =ᶠ[atTop] fun x => p * x ^ (p - 1)
+          := by filter_upwards [eventually_ne_atTop 0] with x hx
+                rw [Real.deriv_rpow_const (Or.inl hx)]
+       _ =Θ[atTop] fun x => x ^ (p-1)
+          := Asymptotics.IsTheta.const_mul_left hp Asymptotics.isTheta_rfl
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean:618]: the formatting looks strange to me here: I would put `:=` on the previous line

**Review by sgouezel — COMMENTED** [review]
LGTM, thanks!
bors d+

**Pull request closed without merging**